### PR TITLE
refactor: template building overhaul

### PR DESCRIPTION
**Pull request by lukeromanowicz**
feat: replace lodash template with ejs
feat: rebuild single files instead of all files in dev mode
feat: add includes in theme
fix: race condition between templates and overwrite components

### CODE REVIEW

**Review by filrak — CHANGES_REQUESTED** [review]
Changes as we discussed - use nuxt module options

**Review by Qrzy — APPROVED** [review]

**Review by andrzejewsky — APPROVED** [review]

File: packages/core/theme-module/ensureDirectoryExists.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+const fs = require('fs');
```
Comment (filrak) [inline packages/core/theme-module/ensureDirectoryExists.js]: maybe lets put all those files into a `scripts` folder. It slowly becomes messy

File: packages/core/theme-module/theme/pages/Category.vue [hunk 2]
```diff
@@ -375,15 +376,13 @@ export default {
...
-    const lastSlug = Object.keys(params).reduce((prev, curr) => params[curr] || prev, params.slug_1);
-
     const { categories, search, loading } = useCategory('categories');
     const { products: categoryProducts, totalProducts, search: productsSearch, loading: productsLoading } = useProduct('categoryProducts');
     const currentPage = ref(parseInt(query.page, 10) || 1);
     const itemsPerPage = ref(parseInt(query.items, 10) || perPageOptions[0]);
 
     onSSR(async () => {
-      await search({ slug: lastSlug });
+      await search(getCategorySearchParameters(context));
```
Comment (filrak) [inline packages/core/theme-module/theme/pages/Category.vue:385]: I reminded you that it should be inlined. `includes` dir should exist only in `theme-module`

**Review by filrak — CHANGES_REQUESTED** [review]

File: packages/core/theme-module/theme/pages/Category.vue (hunk 2 above)
Comment (lukeromanowicz) [reply packages/core/theme-module/theme/pages/Category.vue:385]: We agreed that in the end, we will move this one to an external function

File: packages/core/theme-module/ensureDirectoryExists.js (hunk 1 above)
Comment (lukeromanowicz) [reply packages/core/theme-module/ensureDirectoryExists.js]: fixed

**Pull request merged**